examples/ecies-example.cpp

```diff
@@ -0,0 +1,230 @@
+     * message authentication code, symmetric cipher) used for encryption and decryption.
+     * In the current example the following is used:
+     * - SHA512 everywhere where required
+     * - X963(SHA512) as key derivation function
+     * - HMAC(SHA512) as message authentication code
+     * - AES CBC with PKCS padding, zero IV and 256 bit key size
+     * - Additionally salts for KDF and MAC can be set (optional value, default: empty)
+     *
+     * The defaults if the builder is invoked without any optional parameters are:
+     * - X963(SHA512) as key derivation function
+     * - HMAC(SHA512)
+     * - AES CBC with PKCS padding, zero IV and 256 bit key size
+     * - empty salts for KDF and MAC
+     */
+    const auto digest = DigestTypes::SHA512;
+    const int digestLength = Hash::getDigestSize(digest);
+    const auto kdfFunc = std::make_shared<X963KDF>(digest);
+    /* Dependency injection function, as the key for MAC is generated based on a random ECC key in the ECIES class */
+    const auto macFunc = std::function<std::unique_ptr<MessageAuthenticationCode>(const std::vector<uint8_t>&)>(
+                [digest](const std::vector<uint8_t> &key) -> std::unique_ptr<MessageAuthenticationCode> {
+        return std::make_unique<mococrw::HMAC>(digest, key);
+    });
+
+    /* The symmetric cipher */
+    const SymmetricCipherMode mode = SymmetricCipherMode::CBC;
+    const SymmetricCipherKeySize keySize = SymmetricCipherKeySize::S_256;
+    const SymmetricCipherPadding padding = SymmetricCipherPadding::PKCS;
+    const std::vector<uint8_t> zeroIV = std::vector<uint8_t>(AESCipherBuilder::getDefaultIVLength(mode));
+    const auto aesFunc = [zeroIV](const std::vector<uint8_t> &key) -> std::unique_ptr<SymmetricCipherI> {
+                    return AESCipherBuilder(mode, keySize, key).setIV(zeroIV).setPadding(padding).buildDecryptor();
+                };
+
+
+    /* Read the private key and thet the ECC specification of the private key.
+     * The elliptic curve of the ephemeral key is the same as the private key's one as it is derived from the
+     * corresponding public key. */
+    auto privKey = AsymmetricPrivateKey::readPrivateKeyFromPEM(privKeyPem, "");
+    std::shared_ptr<AsymmetricKey::Spec> spec = privKey.getKeySpec();
+    auto eccSpec = std::dynamic_pointer_cast<ECCSpec>(spec);
+    if (!eccSpec) {
+        std::cerr << "Given key is no ECC key." << std::endl;
+        exit(EXIT_FAILURE);
+    }
+
+    /* The underlying openssl function recoginzes the used serialization format of the ephemeral key reading the first
+     * byte of the serialized data, which defines the format.
+     * The variable eccSpec contains the used elliptic curve of the ephemeral key */
+    auto _ephemeralKey = AsymmetricPublicKey::fromECPoint(eccSpec, eciesData.ephemeralKey);
+
+    /* Get the decryption context */
+    ECIESCtxBuilder decBuilder;
+    auto decCtx = decBuilder
```

Comment:
s.a. regarding inlining the config.
Moreover, there seems to be no need to actually have a named builder object.